### ISSUE TITLE
OSX-ARM64 Investigate Hello World failure.

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/runtime#41119]
With #40435, 

HelloWorld fails like this:
```
   at System.NullReferenceException..ctor()
...
Repeat until stack overflow
...
   at System.NullReferenceException..ctor()
   at System.Array.LastIndexOf[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](System.__Canon[], System.__Canon, Int32, Int32)
   at System.Collections.Generic.List`1[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].LastIndexOf(System.__Canon, Int32, Int32)
   at System.Collections.Generic.List`1[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].LastIndexOf(System.__Canon)
   at System.SR.InternalGetResourceString(System.String)
   at System.SR.GetResourceString(System.String, System.String)
   at System.SR.GetResourceString(System.String)
   at System.SR.get_Arg_NullReferenceException()
   at System.NullReferenceException..ctor()
   at System.Array.LastIndexOf[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](System.__Canon[], System.__Canon, Int32, Int32)
   at System.Collections.Generic.List`1[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].LastIndexOf(System.__Canon, Int32, Int32)
   at System.Collections.Generic.List`1[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].LastIndexOf(System.__Canon)
   at System.SR.InternalGetResourceString(System.String)
   at System.SR.GetResourceString(System.String, System.String)
   at System.SR.GetResourceString(System.String)
   at System.SR.get_Arg_ArrayTypeMismatchException()
   at System.ArrayTypeMismatchException..ctor()
   at System.Runtime.CompilerServices.CastHelpers.StelemRef_Helper_NoCacheLookup(System.Object ByRef, Void*, System.Object)
   at System.Runtime.CompilerServices.CastHelpers.StelemRef_Helper(System.Object ByRef, Void*, System.Object)
   at System.Collections.Generic.ComparerHelpers.CreateDefaultEqualityComparer(System.Type)
   at System.Collections.Generic.EqualityComparer`1[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]..cctor()
   at System.Collections.Generic.EqualityComparer`1[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].get_Default()
   at System.Text.EncodingHelper.GetCharset()
   at System.Text.EncodingHelper.GetCharset()
   at System.Text.EncodingHelper.GetEncodingFromCharset()
   at System.ConsolePal.GetConsoleEncoding()
   at System.ConsolePal.get_OutputEncoding()
   at System.Console.get_OutputEncoding()
   at System.Console.CreateOutputWriter(System.IO.Stream)
   at System.Console.<get_Out>g__EnsureInitialized|25_0()
   at System.Console.get_Out()
   at System.Console.WriteLine(System.String)
   at x64.console.Program.Main(System.String[])
```

**Comment by danmosemsft** [comment]
The amusing thing here is that `InternalGetResourceString` protects itself against recursive lookup - unless it is specifically a failure within `List<T>.LastIndexOf`

One would think that would be rare, but https://github.com/dotnet/runtime/issues/40073 hit it for another reason.

We probably should fix that in `InternalGetResourceString`, not that it would solve the problem here.

**Comment by sdmaclea** [comment]
Fixed in #40435
